Assignment_2/hexadecimalConversion/HexCalc.java

```diff
@@ -0,0 +1,355 @@
+package hexadecimalConversion;
+
+import java.util.Scanner;
+
+/**
+ * HexCalc is a public class which is heavily used in the main class HexCalc_Driver .
+ * 
+ * @author Aman Gautam
+ * 
+ * dated 15/07/2019
+ * 
+ */
+public class HexCalc {
+	private String firstnumber;
+	private String secondnumber;
+	
+	
+	Scanner sc = new Scanner(System.in);
+
+	
+	/**
+	 * this is a getDecimal method for the conversion of HexaDecimal no. to
+	 * Decimal no.
+	 * 
+	 * @param  {String}hexnumber   Hexadecimal no. user input
+	 * 
+	 * @return {integer}value     Decimal equivalent of Hexadecimal no
+	 */
+	public int getDecimal(String hexnumber) {
+		int index, value = 0;
+		char character;
+		String digits = "0123456789ABCDEF", string;
+
+		hexnumber = hexnumber.toUpperCase();
+		for (int i = 0; i < hexnumber.length(); i++) {
+			character = hexnumber.charAt(i);
+			string = Character.toString(character);
+			if (digits.contains(string)) {
+				index = digits.indexOf(character);
+				value = 16 * value + index;
+
+			} else {
+				System.out.println("Invalid Hexadecimal no.");
+				value = 0;
+				break;
+			}
+		}
+		return value;
+
+	}
+
+	
+	/**
+	 * this is check method for checking if hexadecimal no. is valid or invalid
+	 * 
+	 * @param  {String} hexnumber
+	 * 
+	 * @return {boolean} value 
+	 */
+	public boolean check(String hexnumber) {
+		char character;
+		String string;
+		String digits = "0123456789ABCDEF";

```

Comment:
This variable can be defined as a constant

Reply:
const DIGIT defined